test/notifications/display_test.dart

```diff
@@ -131,6 +131,59 @@ void main() {
             NotificationChannelManager.kVibrationPattern)
       ;
     });
+
+    test('channel is not recreated if one with same id already exists', () async {
+      await NotificationChannelManager.ensureChannel();
```

Comment:
It looks like these tests affect the state in the binding, and then leak that state. To fix that, they should start with `addTearDown(testBinding.reset)` (like the existing smoke test does via `init`).